/*
 * Copyright (C) 2025-2026 aisleron.com
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.aisleron.domain.productvariant

import com.aisleron.data.productvariant.ProductWithBarcode
import com.aisleron.domain.base.BaseRepository
import kotlinx.coroutines.flow.Flow

interface ProductVariantRepository : BaseRepository<ProductVariant> {
    fun getByBarcode(barcode: String): Flow<ProductVariant?>
    fun getByProductId(productId: Int): Flow<List<ProductVariant>>
    fun barcodeExists(barcode: String): Flow<Boolean>
    suspend fun deleteByBarcode(barcode: String)
    fun getProductWithBarcode(barcode: String): Flow<ProductWithBarcode?>
}
